crates/ruff_workspace/src/configuration.rs

```diff
@@ -1132,6 +1135,201 @@ pub fn resolve_src(src: &[String], project_root: &Path) -> Result<Vec<PathBuf>>
     Ok(paths)
 }
 
+fn warn_about_deprecated_top_level_lint_options(top_level_options: &LintCommonOptions) {
```

Comment:
This feels a bit silly, but writing a macro for it didn't feel worth the time because we shouldn't be adding any new options to `LintCommonOptions` anyway.